sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HttpRequestLogger.java

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.http.policy;
+
+import com.azure.core.http.HttpPipelineCallContext;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.core.util.logging.LogLevel;
+import reactor.core.publisher.Mono;
+
+/**
+ * Manages logging HTTP requests in {@link HttpLoggingPolicy}.
+ */
+public interface HttpRequestLogger {
+    /**
+     * Retrieves the {@link LogLevel} used to log the current request.
+     *
+     * @param defaultLogLevel The default log level to use.
+     * @param callContext The contextual information about the request, including headers, body, and metadata.
+     * @return The {@link LogLevel} used to log the current request.
+     */
+    LogLevel getLogLevel(LogLevel defaultLogLevel, HttpPipelineCallContext callContext);
```

Comment:
Do we need this method? For anyone implementing this interface, they can simply use the `logRequest` method to log at whatever level they want. We can then make this a functional interface.

Reply:
It is meant to cover scenarios where each request could have a different log level. This could be a default interface method that just returns `LogLevel.INFORMATIONAL`, allowing for it to effectively be a functional interface.